googleapiclient/http.py

```diff
@@ -56,6 +56,10 @@
 from googleapiclient.model import JsonModel
 from oauth2client import util
 
+try:
```

Comment:
On Windows, in Python 2.7.11, a network timeout looks like this:
```
  File "C:\Python27_11\lib\site-packages\httplib2\__init__.py", line 1609, in request
    (response, content) = self._request(conn, authority, uri, request_uri, method, body, headers, redirections, cachekey)
  File "C:\Python27_11\lib\site-packages\httplib2\__init__.py", line 1351, in _request
    (response, content) = self._conn_request(conn, request_uri, method, body, headers)
  File "C:\Python27_11\lib\site-packages\httplib2\__init__.py", line 1272, in _conn_request
    conn.connect()
  File "C:\Python27_11\lib\site-packages\httplib2\__init__.py", line 928, in connect
    raise socket.error, msg
error: [Errno 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
```

I tried catching `socket.timeout` and it doesn't catch the exception (as expected from this trace), so I believe it's better to retry on a non-retriable error than miss a retriable one - especially one as important as a network timeout.